src/plugins/dump/dump.cpp

```diff
@@ -78,125 +111,347 @@ int serialise (std::ostream & os, ckdb::Key *, ckdb::KeySet * ks)
 				/* Meta key already serialised, write out a reference to it */
 				keyDel (search);
 
-				os << "keyCopyMeta ";
-				os.write (static_cast<const char *> (ckdb::keyValue (ret)), ckdb::keyGetValueSize (ret));
-				os << std::endl;
+				os << "$copymeta " << keyString (ret);
 			}
 		}
-		os << "keyEnd" << std::endl;
-	}
-	os << "ksEnd" << std::endl;
 
+		os.flush ();
+	}
 	ksDel (metacopies);
 
+	if (markEnd)
```

Comment:
I don't like that it is sometimes written and sometimes not. I would prefer if it is always written. Then streaming always works.

Reply:
You are right, it is probably best, if we always write it. But I'll keep the read part, such that it is optional and only stops processing of file early.